### PR TITLE
fix: disable button when value is empty

### LINKED ISSUE #3534: "Save" button should be disabled when there's no name given to the New Collection
When the name filed is blank, i.e. before the user entering a name for the New Collection, the visual appearance and function of the Save button should be disabled. Currently, it's clickable with no effect.

image

### PR DESCRIPTION
**Pull request by Wzb3422**
Fixes #3534 

This pull request resolves the [issue#3534](https://github.com/toeverything/AFFiNE/issues/3534)

### CODE REVIEW

**Review by himself65 — CHANGES_REQUESTED** [review]
useMemo here

File: packages/component/src/components/page-list/view/create-collection.tsx [hunk 1]
```diff
@@ -274,9 +274,9 @@ export const EditCollection = ({
           size="large"
           data-testid="save-collection"
           type="primary"
-          disabled={isNameEmpty(value)}
+          disabled={isNameEmpty}
           onClick={() => {
```
Comment (himself65) [inline packages/component/src/components/page-list/view/create-collection.tsx]: Could you please use useCallback here? Thx

**Comment by Wzb3422** [comment]
> useMemo here

Sure. `useMemo` does a better job in memorization. I've refactored my code with `useMemo`.

File: packages/component/src/components/page-list/view/create-collection.tsx (hunk 1 above)
Comment (Wzb3422) [reply packages/component/src/components/page-list/view/create-collection.tsx]: Done. Refactored this implementation with `useCallback`.

**Review by himself65 — APPROVED** [review]

**Pull request merged**